### ISSUE TITLE
Static Caching and Page Reload method

### ISSUE THREAD

**Issue by chrisrowe** [verbb/formie#1993]
### Describe the bug

When using the Page Reload submission method we're returning the same page at the same url the form hidden and the success message, this this url is being cached and future visits don't show the form.
The usual `Formie.refreshForCache('{{ form.formId }}');` method doesn't work here and doesn't restore the form and hide the success message

### Steps to reproduce

1. Clear all caches
2. Prepare a front end form for submission
3. Enable static caching
4. Submit the form with Page Reload method
5. Cache gets warmed with sucess message only

### Form settings

- Multi-page form: No
- Submission Method: Page Reload
- Client-side Validation: No
- Custom Form Templates: No

### Craft CMS version

4.9.3

### Plugin version

2.1.14

### Multi-site?

No

### Additional context

We're hosted on Servd using static caching

**Comment by engram-design** [comment]
Using a static cache with Page Reload is indeed going to cause some issues. But I would have expected the page to be cached as soon as you visit the form. When the form is submitted, you would typically see that same page again. Are you saying on the next page load (after submitting the form and seeing the success message), it's the final success message you see cached?

Your options are to use Ajax for form submissions, exclude the page with the form from cache, or use another URL (or add a query string to the current one) as the success page, which will be cached separately.

**Comment by chrisrowe** [comment]
Yeah, the problem is if the cache is cleared and then someone submits the form the warmed version is the success page.

We have a client with hundreds of forms so I was trying to avoid the switching to Ajax method if possible, excluding from cache isn't easy either since we're not aware of every URL they're on up front. Ideally we'd have the Page Reload append a querystring but I don't actually see a way to do that?

**Comment by engram-design** [comment]
There's a few issues floating around here to the same effect about trying to address this. Indeed, we'll probably need to append a unique query string to each page submission, but there's not a way to do that for multi-page forms. For single-page forms, you could have it so that the success page contains a query string in the form settings.

It's something we want to address, likely by adding a `staticCacheMode` plugin setting or similar to just automate all the various bits of functionality required for a statically-cached site.

**Comment by chrisrowe** [comment]
Sounds good! Yeah something like `mysite.com/contact?formHandle=page1` for multi-page and just `/contact?formHandle=submitted` for everything else would be perfect.

I guess for now I can use the `afterSubmissionRequest` and append my own querystring to the redirect url?

**Comment by chrisrowe** [comment]
For now I came up with this which seems to do the job.

```php
Event::on(
    SubmissionsController::class,
    SubmissionsController::EVENT_AFTER_SUBMISSION_REQUEST,
    function(SubmissionEvent $event) {
        $form = $event->form;
        $submission = $event->submission;
        $success = $event->success;

        if (!$success) {
            return;
        }

        if (
            $form->settings->submitMethod == "page-reload" &&
            $form->settings->submitAction == 'message' &&
            $form->settings->redirectUrl == null            
        ) {

            Formie::$plugin->getService()->setFlash($form->id, 'submitted', true);
            Formie::$plugin->getService()->setNotice($form->id, $form->settings->getSubmitActionMessage($submission));

            Craft::$app->getUrlManager()->setRouteParams([
                'submission' => $submission,
            ]);

            $url = UrlHelper::urlWithParams(Craft::$app->request->getUrl(), [
                'success' => true,
                'rng' => Craft::$app->security->generateRandomString(10),
            ]);
            Craft::$app->response->redirect($url)->send();

        }
    }
);
```

**Comment by engram-design** [comment]
Updated in [4.0.0-beta.1](https://github.com/verbb/formie/releases/tag/4.0.0-beta.1).
